cli/azd/cmd/up.go

```diff
@@ -146,6 +146,8 @@ func (u *upAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Use the correlation id from the provision operation
+	deployResult.Message.CorrelationId = provisionResult.Message.CorrelationId
```

Comment:
This feels wrong to me at some level - the deploy operation is going to do a bunch of work (deploying each service) and each piece of work logically has its own correlation ID.  I think long term we would want to aggregate all the correlation ids generated by each step of the `up` flow and display them all.

I suspect that this is the path of least resistance in order for us to display the correlation id from the provision operation at the end of the `up` flow - so I'm not going to block on this feedback, but I think we need to think long term about how we want to handle this sort of thing in the up flow.

Honestly, I'm not sold on the value of printing these correlation IDs to the end user on all these operations.  My mental model is that these are not useful except in the cases where I want to open a support ticket.  I could see printing them on error paths (as part of context of the error), and perhaps `log.Print`'ing them in all cases (since if you are running with `--debug, they could be useful), but I do feel like the UX of the tool got a little worse with this change when viewed through the lens of someone who is new to Azure.

In practice, it seems like you can [recover the correlation id from the deployment itself](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/deployment-history?tabs=azure-portal) (which is retained), so it's not as if dropping this value on the floor makes it unrecoverable.

Reply:
I didn't want to have the correlatioID on happy paths as well.  For example, on success cases.
Rajesh said that sometimes, even there is a success message, things might be working, and then we need that Id for creating the ticket.

I am just happy that you and Wei show up here and we'll be removing this correlationId from happy paths :)

Reply:
@ellismg @vhvb1989 
CSS was asking for Azd to support/enable log files. So by default, we have debug logs with rotation policies in place.

Was thinking there would one only one correlation-if for our deployment. If more than one, should we instead log correlation id's to logs?

Reply:
> CSS was asking for Azd to also support/enable log files. So by default, we have debug logs with rotation policies in place. Thought there would one only one correlation-if for our deployment. If more than one, should we instead log correlation id's to logs?

After reading a little more, I think what we should be doing is something like this (to be clear, I'm not suggesting we do this as part of this PR)

1. When `azd` starts up, we should generate a random UUID, and add it to our context (we should also `debug.Log` it)
2. We should make the necessary changes to our clients such that we set an `x-ms-correlation-request-id` header on all the requests we make (this allows multiple independent requests to be correlated back to a single run of the CLI).
3. We should update our logging policy to add this header to the list of allowed headers to log (again, we want to do this globally, for all clients)

I think with these changes, we'll be logging both the client specific request ID (which is generated per operation) as well as an ID which can be used to correlate multiple REST calls across a single AZD operation.

This should give us the complete set of IDs that CSS can use to investigate.

I'm guessing at this based on the discussion in https://github.com/Azure/azure-sdk/issues/576 and the linked documentation that describes how ARM uses these values and flows them to downstream RPs.

But, @weikanglim is the expert here, so he can check my work.

Reply:
IIRC, the requestId is automatically generated by the Azure SDK Core pipeline.

Reply:
> IIRC, the requestId is automatically generated by the Azure SDK Core pipeline.

It generates a unique ID per request, which it sets as `x-ms-client-request-id`, the `x-ms-correlation-request-id` header can be used to provide an ID which is the same across multiple requests which represent a logical operation (it sort of feels like a `traceparent` in distributed tracing parlance).  I think this makes it easier for support folks to do queries like: "show me all the requests that happened across a single invocation of `azd provision`)

Reply:
@ellismg You're indeed correct. We already have a `traceId` generated by default in our root span. We would use it to set  `x-ms-correlation-request-id`.  I updated #992